packages/block-editor/src/store/selectors.js

```diff
@@ -148,9 +163,13 @@ export const getBlock = createSelector(
 		return {
 			...block,
 			attributes: getBlockAttributes( state, clientId ),
-			innerBlocks: areInnerBlocksControlled( state, clientId )
-				? EMPTY_ARRAY
-				: getBlocks( state, clientId ),
+			innerBlocks:
+				! includeControlledInnerBlocks &&
+				areInnerBlocksControlled( state, clientId )
+					? EMPTY_ARRAY
+					: getBlocks( state, clientId, {
+							includeControlledInnerBlocks,
+					  } ),
```

Comment:
This call also breaks memoization as it will recompute getBlocks on each call since we're passing and inline object.